### ISSUE TITLE
Color management clarification.

### ISSUE THREAD

**Issue by panzergame** [UPBGE/blender#760]
Currently as blender is directly rendering to the screen the shaders include the conversion to SRGB space, but as UPBGE is rendering to a frame buffer which is later "copied" to the screen, we could apply the space conversion only at the end and remove it from the material shader.

I just have one question about the side effects: By doing this the filters will see as input the color in RGB space instead of SRGB as possibly before.
Also custom shaders will be altered by SRGB contrary to before.

@lordloki, @youle31, @martinsh : What are your opinion ?

**Comment by youle31** [comment]
Hello, in eevee pipe, filters are rendered before the tonemapping (DRW_transform_to_display). This scheme can be applied to upbge rendering pipeline I think (but I can be wrong). Else, in bge there is sometimes banding effects on the final rendered image. This can be fixed adding dithering (noise). I think this can be fixed even if the image is RGBA8. This would improve final image quality but needs an artist and a large screen to test.

**Comment by adriansnetlis** [comment]
I would totally love linear RGB colors in nodes. This is how I currently tonemap (because every sensible human being tonemaps using filmic instead of raw sRGB):
- Gamma scale 1.0/2.2 (a.k.a. sRGB to linear)
- Any other filters
- Filmic tonemap

The first step could be removed as a result, meaning that 2 steps would be taken away from the rendering process.

**Comment by adriansnetlis** [comment]
To clarify - your proposal is totally right and makes 100% sense (to be honest, I didn't know that sRGB conversion happened inside shaders, that is very unusual and weird way to do it). However, having sRGB applied to frame buffer should be optional. There should be option for linear, sRGB and maybe some other color management schemes (probably even some sort of filmic tonemap implementation).

**Comment by panzergame** [comment]
@adriansnetlis : Is it needed to apply filters after the color space conversion ?

**Comment by adriansnetlis** [comment]
No. It makes more sense if filters are applied before color management (giving them access to raw color data which is much more usable). Color management is basically always the last thing on the list.
